tests/Unit/Helpers/Elliptic/FirstOrderSystem.hpp

```diff
@@ -0,0 +1,318 @@
+  tmpl::for_each<argument_tags>(
+      [&args, &generator, &dist, &used_for_size](auto tag_v) {
+        using tag = tmpl::type_from<decltype(tag_v)>;
+        get<tag>(args) = make_with_random_values<typename tag::type>(
+            make_not_null(&generator), make_not_null(&dist), used_for_size);
+      });
+  const auto expected_sources = tuples::apply(
+      [&vars, &fluxes](const auto&... expanded_args) {
+        return ::elliptic::first_order_sources<
+            volume_dim, primal_fields, auxiliary_fields, SourcesComputer>(
+            vars, fluxes, expanded_args...);
+      },
+      args);
+
+  // Create a DataBox
+  auto box = tuples::apply(
+      [&vars, &fluxes, &used_for_size](const auto&... expanded_args) {
+        return db::create<
+            tmpl::append<db::AddSimpleTags<vars_tag, fluxes_tag, sources_tag>,
+                         argument_tags>>(
+            vars, fluxes,
+            make_with_value<typename sources_tag::type>(
+                used_for_size, std::numeric_limits<double>::signaling_NaN()),
+            expanded_args...);
+      },
+      args);
+
+  // Apply the sources computer to the DataBox
+  tuples::tagged_tuple_from_typelist<
+      tmpl::append<primal_fields,
+                   db::wrap_tags_in<::Tags::Flux, primal_fields,
+                                    tmpl::size_t<volume_dim>, Frame::Inertial>>>
+      vars_tuple{};
+  tmpl::for_each<primal_fields>([&vars, &fluxes, &vars_tuple](auto tag_v) {
+    using tag = tmpl::type_from<decltype(tag_v)>;
+    using flux_tag =
+        ::Tags::Flux<tag, tmpl::size_t<volume_dim>, Frame::Inertial>;
+    get<tag>(vars_tuple) = get<tag>(vars);
+    get<flux_tag>(vars_tuple) = get<flux_tag>(fluxes);
+  });
+  tmpl::for_each<auxiliary_fields>([&vars, &box](auto tag_v) {
+    using tag = tmpl::type_from<decltype(tag_v)>;
+    db::mutate<::Tags::Source<tag>>(
+        make_not_null(&box),
+        [&vars](const auto aux_source) { *aux_source = get<tag>(vars); });
+  });
+  tuples::apply(
+      [&box](const auto&... expanded_vars) {
+        db::mutate_apply<
+            db::wrap_tags_in<::Tags::Source, typename vars_tag::tags_list>,
+            argument_tags>(SourcesComputer{}, make_not_null(&box),
+                           expanded_vars...);
+      },
+      vars_tuple);
+  CHECK(expected_sources == get<sources_tag>(box));
+}
+
+namespace detail {
+template <typename Tag>
+struct Correction : db::PrefixTag, db::SimpleTag {
+  using type = typename Tag::type;
+  using tag = Tag;
+};
+template <typename Tag>
+struct Corrected : db::PrefixTag, db::SimpleTag {
+  using type = typename Tag::type;
+  using tag = Tag;
+};
+}  // namespace detail
+
+/// Test the `System` has a `linearized_system` and that it is actually the
+/// linearization of its nonlinear fluxes and sources to the order given by
+/// the `correction_magnitude`
+template <typename System>
+void test_linearization(const double correction_magnitude,
```

Comment:
Any chance I could convince you to either include a use of this test helper in this PR or postpone its inclusion to a PR that will use it? I'm generally dubious of merging in code that isn't yet run at all in the present tests or executables, even if it's in a test helper (doubly so if it isn't actually compiled as with uninstantiated templates ;))

Reply:
I agree with your dislike of merging code that isn't compiled :D None of the existing systems are nonlinear, so I can't really use it in an existing test. I could move this code back to #2667, but the reason this was moved here was to reduce the size of the XCTS PR, and to make it all about the math instead of metaprogramming things. So I would suggest to just merge this test helper uncompiled, knowing that #2667 compiles it 🤷‍♂️

Reply:
We have also quite often written tests for our helpers. I think a lot of the domain helpers have this, and I know I've done it for the boundary corrections tests

Reply:
I don't think adding a test for this test helper is worth the effort.. it's probably more code than the test helper itself

Reply:
tests can often be longer than the code they test -- regardless, I think I feel more strongly about the principal of 'don't merge code that isn't compiled' than I want to save the ~100 lines from the future PR. So, my vote is still to either add a test that proves basic functionality, or move it back to the PR it got split out from.

Reply:
Ok moved the code back to #2667